c7n/resources/glacier.py

```diff
@@ -157,3 +157,30 @@ def process_resource(self, client, resource):
         return {'Name': resource['VaultName'],
                 'State': 'PolicyRemoved',
                 'Statements': found}
+
+
+@Glacier.action_registry.register('delete')
+class GlacierVaultDelete(RemovePolicyBase):
+    """Action to delete glacier vaults
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: glacier-vault-delete
+                resource: aws.glacier
+                filters:
+                  - type: cross-account
+                actions:
+                  - type: delete
+    """
+
+    schema = type_schema('delete')
+    permissions = ('glacier:DeleteVault',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('glacier')
+        for r in resources:
+            self.manager.retry(client.delete_vault, vaultName=r['VaultName'], ignore_err_codes=(
```

Comment:
A glacier vault deletion will fail if it has archives since the last inventory. Could we handle that error and delete archives then delete the false using something like a force boolean? Similar to what we do with some other resources with delete protection.

Reply:
the concern is that we're getting into data plane, ala deleting an s3 bucket which has objects is a similar analogy, people will end up using it on containers with billions of objects ;-) and trying to clean it ends up having some repurcussions on runtime.

Reply:
yea, that was mostly why I did not want to add a force option. @JohnHillegass for our use case we are only trying to delete net new glacier vaults which is achievable with this

Reply:
Fair enough regarding data plane and focusing mainly on net-new resource coverage. If we ever get into a need for existing it may make sense to add but probably not now.